model/QueueBroker/NewSqlQueueBroker.php

```diff
@@ -0,0 +1,271 @@
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SchemaException;
+use Exception;
+use InvalidArgumentException;
+use oat\generis\Helper\UuidPrimaryKeyTrait;
+use oat\generis\persistence\PersistenceManager;
+use oat\tao\model\taskQueue\Queue\Broker\AbstractQueueBroker;
+use oat\tao\model\taskQueue\Task\TaskInterface;
+use oat\taoTaskQueue\model\QueueBroker\storage\NewSqlSchema;
+use Throwable;
+
+/**
+ * Storing messages/tasks in DB.
+ *
+ * @author Gyula Szucs <gyula@taotesting.com>
+ */
+class NewSqlQueueBroker extends AbstractQueueBroker
+{
+    use UuidPrimaryKeyTrait;
+
+    private $persistenceId;
+
+    /** @var common_persistence_SqlPersistence */
+    protected $persistence;
+
+    public function __construct(string $persistenceId, int $receiveTasks = 1)
+    {
+        parent::__construct($receiveTasks);
+
+        if (empty($persistenceId)) {
+            throw new InvalidArgumentException("Persistence id needs to be set for " . __CLASS__);
+        }
+
+        $this->persistenceId = $persistenceId;
+    }
+
+    public function __toPhpCode()
+    {
+        return 'new ' . get_called_class() . '('
+            . \common_Utils::toHumanReadablePhpString($this->persistenceId)
+            . ', '
+            . \common_Utils::toHumanReadablePhpString($this->getNumberOfTasksToReceive())
+            . ')';
+    }
+
+    protected function getPersistence(): ?common_persistence_SqlPersistence
+    {
+        if (is_null($this->persistence)) {
+            $this->persistence = $this->getServiceLocator()
+                ->get(common_persistence_Manager::SERVICE_ID)
+                ->getPersistenceById($this->persistenceId);
+        }
+
+        return $this->persistence;
+    }
+
+    private function getTableName(): string
+    {
+        return strtolower($this->getQueueNameWithPrefix());
+    }
+
+    /**
+     * Note: this method can be run multiple times because only the migrate queries (result of getMigrateSchemaSql) will be run.
+     *
+     * @inheritdoc
+     */
+    public function createQueue()
+    {
+        $persistence = $this->getPersistence();
+        /** @var AbstractSchemaManager $schemaManager */
+        $schemaManager = $persistence->getDriver()->getSchemaManager();
+
+        /** @var Schema $schema */
+        $schema = $schemaManager->createSchema();
+        $fromSchema = clone $schema;
+
+        if (in_array($this->getTableName(), $schemaManager->getTables())) {
+            $schema->dropTable($this->getTableName());
+        }
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+
+        try {
+            $this->getSchemaProvider($schema)
+                ->setQueueName($this->getQueueName())
+                ->getSchema($schema, $this->getTableName());
```

Comment:
```suggestion
            $schema = $this->getSchemaProvider($schema)
                ->setQueueName($this->getQueueName())
                ->getSchema($schema, $this->getTableName());
```

It is getting a new schema but never using it. It is correct?